app/src/main/java/com/example/lunark/HomeActivity.java

```diff
@@ -57,16 +67,39 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         actionBarDrawerToggle.syncState();
 
         findViewById(R.id.filterButton).setOnClickListener(v -> {
-            FilterDialogFragment filterDialog = new FilterDialogFragment();
-            FragmentManager fragmentManager = getSupportFragmentManager();
-            filterDialog.show(fragmentManager, "filter_dialog");
-                }
-
+                new FiltersDialogFragment().show(
+                        getSupportFragmentManager(), FiltersDialogFragment.TAG
+                );
```

Comment:
If you click the filter button fast, multiple filter dialogs will appear. I suggest you throttle this listener to prevent this from happening.
You can fix this by:
- [Throttling the listener](https://stackoverflow.com/questions/31868874/fast-taps-clicks-on-recyclerview-opens-multiple-fragments)
- [Checking if the dialog is shown by having a reference to the dialog in HomeActivity](https://stackoverflow.com/a/12370187)
- [Using a boolean variable to check if a click listener is being ran](https://stackoverflow.com/a/31868975)

Any of the above methods would be appropriate.